.docs/concepts.md

```diff
@@ -0,0 +1,96 @@
+# Application Insights DotNet SDK Concepts
+This lists the high level concepts of our SDK and links to detailed guides to help you get started.
+
+
+To use the Application Insights SDK you must configure an Instrumentation Key which can be [obtained from an Application Insights resource](https://docs.microsoft.com/azure/azure-monitor/app/create-new-resource).
+Or you can use a [Connection string](https://docs.microsoft.com/azure/azure-monitor/app/sdk-connection-string?tabs=net) to simplify the configuration of our endpoints.
+Both an Instrumentation Key and Connection String are provided for you on the Overivew Dashboard of your Application Insights resource.
+
+
+
+## TelemetryClient
+The `TelemetryClient` object is the primary root object for the library. 
+Almost all functionality around telemetry sending is located on this object. 
+
+### Initialization
+You must initialize an instance of this object and populate it with your Instrumentation Key to identify your data.
+
+```C#
+using Microsoft.ApplicationInsights;
+using Microsoft.ApplicationInsights.Extensibility;
+
+var configuration = new TelemetryConfiguration
+{
+    // Please use either an Instrumentation Key or a Connection String but not both.
+    InstrumentationKey = "YOUR INSTRUMENTAATION KEY",
+    ConnectionString = "YOUR CONNECTION STRING",
+};
+var tc = new TelemetryClient(configuration);
+```
+
+### Using the TelemetryClient to send telemetry
+You can populate common context on the `TelemetryClient.context` property which will be automatically attached to each telemetry item sent. 
+You can also attach additional property data to each telemetry item sent. 
+The `TelemetryClient` also exposes a number of `Track` methods that can be used to send all telemetry types understood by the Application Insights service. Some example use cases are shown below.
+
+```C#
+tc.Context.User.Id = Environment.GetUserName(); // This is probably a bad idea from a PII perspective.
+tc.Context.Device.OperatingSystem = Environment.OSVersion.ToString();
+
+tc.TrackPageView("Form1");
+
+tc.TrackEvent("PurchaseOrderSubmitted", new Dictionary<string, string>() { {"CouponCode", "JULY2015" } }, new Dictionary<string, double>() { {"OrderTotal", 68.99 }, {"ItemsOrdered", 5} });
+	
+try
+{
+	...
+}
+catch(Exception e)
+{
+	tc.TrackException(e);
+}
+``` 
+
+### TelemetryClient API
+The TelemetryClient provides several Track methods to collect different types of logs
+
+Please review our [Api summary for custom events and metrics](https://docs.microsoft.com/azure/azure-monitor/app/api-custom-events-metrics).
+
+
+
+## Telemetry Channels
+Telemetry channels are responsible for buffering telemetry items and sending them to the Application Insights service, where they're stored for querying and analysis.
```

Comment:
Not necessary to have `buffering`?

Reply:
I didn't write any of these statements. That's an exact quote from our public doc. Do you think it should say something else here?

Reply:
We can kill buffering in the public docs as well if needed. A lot of the core terminology used in the public docs predates all of us working on the product so if there is something that doesn't make sense, definitely let me know so we can clean it up.

Reply:
I wonder if this would help to clarify "Telemetry channels are responsible for sending the telemetry data to the designated place. Optional features can be provided by the telemetry channels, for example, buffering the data and sending in them in batches, persisting the data to a local storage in case of transmission failure (e.g. network outage), traffic shaping and retry mechanisms".

Reply:
How about:
"Telemetry channels are responsible for sending telemetry data. Optional features include buffering and batching data, local data storage in case of transmission failure, and retry mechanisms."

Reply:
Why dont we just link to public doc here? for every topic where there is a public doc, i'd suggest to simply link it here and not write anything more.

Reply:
@cijothomas, i've given every section an intro statement, copied from the public doc and also a link to that public doc. 
The purpose is if someone knew nothing about the sdk, they could navigate high-level concepts all in one place.
Similar to this: https://github.com/microsoft/ApplicationInsights-Home/issues/256